modules/@apostrophecms/notification/index.js

```diff
@@ -159,14 +167,17 @@ module.exports = {
 
       async find(req, options) {
         try {
-          const notifications = await self.db.find({
+          const results = await self.db.find({
             userId: req.user._id,
-            ...(options.modifiedOnOrSince && { createdAt: { $gt: new Date(options.modifiedOnOrSince) } })
+            ...(options.modifiedOnOrSince && { updatedAt: { $gt: new Date(options.modifiedOnOrSince) } })
```

Comment:
We're getting closer!

Now take a look at the updated design in which I've called for the value of `modifiedOnOrSince` to also include information about the _ids that we have already received that have that `updatedAt` value, so the server can filter them out with `$nin` and not send them again.

Then we can use `$gte`, not `$gt`, and make sure we do eventually get every notification even if two have the same timestamp somehow.

Also it is important that we use `$currentDate`. That means changing our insert operation to use a mongodb "upsert". That allows the update method to be used to do the insertion, which allows the use of `$currentDate`. The purpose of this is to make sure we're not looking at slightly different timestamps from two appservers - only at a consistent database server time.

Reply:
Ok.